.github/workflows/prerelease.yml

```diff
@@ -1,69 +1,71 @@
-      - uses: ./.github/actions/setup
+      - name: Run Setup
+        uses: ./.github/actions/setup
         with:
-          npm_token: ${{secrets.npm_token}}
+          npm_token: ${{ secrets.npm_token }}
       - uses: ./.github/actions/git-creds
       - name: Check if there are changed packages
-        id: changed-packages
         run: |
           changed_packages=$(yarn -s lerna ls --since=origin/master --json --loglevel=error)
-          [[ $changed_packages = "[]" ]] && echo "No packages changed in this branch" && exit 1 || echo "Continuing to prerelease"
+          [[ $changed_packages = "[]" ]] && echo "Skipping prerelease. Nothing to prerelease as Lerna didn't detect any changes." && exit 1 || echo "Changed detected. Continuing to prerelease" && echo "$changed_packages"
       - name: Build all affected packages
-        run: yarn lerna run build --since=origin/master
-      - name: Generate prerelease prefix
-        run: |
-          normalized_branch_name=$(git rev-parse --abbrev-ref HEAD | tr -dc '[:alnum:]\n\r' | tr '[:upper:]' '[:lower:]')
-          echo "BRANCH=$normalized_branch_name" >> $GITHUB_ENV
+        run: yarn lerna run build --since=origin/master --include-dependencies
+      - name: Run Tests
+        shell: bash
+        run: yarn lerna run test --since=origin/master
       - name: Generate new versions
-        run: yarn lerna version --conventional-commits --conventional-prerelease --no-changelog --no-push --preid $BRANCH -y
+        id: generate-versions
+        run: |
+          preid=${{ github.ref_name }}
+          normalized_pre_id=$(echo "$preid" | tr -dc '[:alnum:]\n\r' | tr '[:upper:]' '[:lower:]')
+
+          full_output=$(yarn lerna version --conventional-commits --conventional-prerelease --json --no-changelog --no-push --preid "$normalized_pre_id" -y)
+
+          # extract only the JSON portion from the full output, excluding yarn logs
+          json_output=$(echo "$full_output" | awk '/^\[/{p=1} p; /\]/{p=0}' )
+
+          echo 'new_versions<<EOF' >> $GITHUB_OUTPUT
+          echo "$json_output" >> $GITHUB_OUTPUT
+          echo 'EOF' >> $GITHUB_OUTPUT
       - run: yarn config set registry https://registry.npmjs.org/
       - name: Setup .npmrc for publish
         run: npm set "//registry.npmjs.org/:_authToken=$NODE_AUTH_TOKEN"
         env:
-          NODE_AUTH_TOKEN: ${{secrets.npm_token}}
-      - run: npm whoami
+          NODE_AUTH_TOKEN: ${{ secrets.npm_token }}
       - name: Publish new versions
-        run: yarn lerna publish from-package --dist-tag $BRANCH -y
+        run: yarn lerna publish from-package --dist-tag prerelease -y
       - name: Get current PR id
-        uses: 8BitJonny/gh-get-current-pr@2.2.0
+        uses: 8BitJonny/gh-get-current-pr@3.0.0
         id: PR
-        if: success()
-      - name: Create comment with prerelease version details
-        if: success()
-        uses: peter-evans/create-or-update-comment@v1
+      - name: Prepare PR comment
```

Comment:
Why all the changes to these steps? It feels like we're not gaining a lot, but it makes them much much more complicated to understand, debug and maintain.

Reply:
To keep same functionality. we used to say which version is published, currently it points to the wrong versioning (only branch name without version and prerelease patch count), also now it says which version was published.

Reply:
Which packages*

Reply:
Got ya, although it's important to state that it points to the correct tag, you don't have to have the full version.

Reply:
Yep, but in case we'll have 2 prereleases (though it's rare at Vibe, we only use it at certain cases) or for future reference

Reply:
I wanna make sure that it's understood I'm not against the current implementation, though 👀 just wanted it to have the same as it had till today